packages/flutter/lib/src/painting/text_style.dart

```diff
@@ -6,6 +6,7 @@
 import 'dart:ui' as ui show ParagraphStyle, TextStyle, StrutStyle, lerpDouble, Shadow, FontFeature, TextHeightBehavior, TextLeadingDistribution;
 
 import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart' show TextOverflow;
```

Comment:
I'm not sure how to prevent this dependency loop causing analyzer tests to fail: https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket.appspot.com/8851862873626269600/+/steps/prepare_environment/0/steps/Analyze/0/steps/analyze/0/stdout

![Screen Shot 2021-03-24 at 19 36 48](https://user-images.githubusercontent.com/33294549/112365750-4a29a580-8cd8-11eb-869b-7d2492a1cc11.png)

Reply:
Could we move the TextOverflow enum into painting?  CC @goderbauer

Reply:
Sounds reasonable. May have to re-export it from rendering to avoid breakages.

Reply:
> May have to re-export it from rendering to avoid breakages.

Sorry I quite don't get what I have to do. Should I move TextOverflow into /painting? in a new or existing file?

Reply:
Yeah try moving TextOverflow to /painting.  I think text_painter.dart is probably best at first glance.

Then we'll see if anything that was trying to import TextOverflow fails.  If so then rendering may need to import it from painting and then re-export it again.

Reply:
I think that worked out :)